packages/shared/lib/core/wallet/actions/mintNft.ts

```diff
@@ -0,0 +1,51 @@
+import { get } from 'svelte/store'
+import { createAliasIfNecessary, selectedAccount, updateSelectedAccount } from '@core/account'
+import { NftOptions, TransactionOptions } from '@iota/wallet'
+import { activeProfile, ProfileType } from '@core/profile'
+import { handleLedgerError } from '@core/ledger'
+import { addActivityToAccountActivitiesInAllAccountActivities, resetMintNftDetails } from '../stores'
+import { showAppNotification } from '@lib/notifications'
+import { localize } from '@core/i18n'
+import { Activity } from '../classes'
+import { preprocessTransaction } from '../utils'
+import { Converter } from '@lib/converter'
+import { INftMetadata } from '../interfaces'
+
+export async function mintNft(metadata: INftMetadata): Promise<void> {
+    try {
+        updateSelectedAccount({ isTransferring: true })
+        const account = get(selectedAccount)
+        await createAliasIfNecessary(account)
+
+        const nftOptions: NftOptions = {
+            immutableMetadata: Converter.utf8ToHex(JSON.stringify(metadata), true),
+        }
+        const transactionOptions: TransactionOptions = {
+            remainderValueStrategy: { strategy: 'ReuseAddress', value: null },
+        }
+
+        const mintNftTransaction = await account.mintNfts([nftOptions], transactionOptions)
+        // QUESTION: do we need a buildPersistedAssetFromIrc27Metadata() then addPersistedAsset()?
```

Comment:
`buildPersistedAssetFromIrc30Metadata()` and `addPersistedAsset` are used for native tokens so that we dont need to fetch the metadata everytime we log in.

i think in the future we need to include some similar caching for nft metadata (and then later even somehow cache the  file content). So you can remove the comment and create a task